libshaderc/include/shaderc.h

```diff
@@ -31,6 +32,13 @@ typedef enum {
   shaderc_glsl_tess_evaluation_shader
 } shaderc_shader_kind;
 
+typedef enum {
+  shaderc_target_env_vulkan,
+  shaderc_target_env_glsl,
+  shaderc_target_env_glsl_compat,
```

Comment:
I've added documentation for each target environment here.

In a nutshell - the compat mode allows use of compatibility profile built-in functions (e.g. texture2D). To see the list, grep for "spv == 0" in glslang/MachineIndependent/Initialize.cpp.

I agree it's not clearly defined, but IMO this is still useful functionality to expose. This matches what Shaderc accepted prior to commit 68b4a04.